apps/fluent-tester/src/E2E/common/BasePage.win.ts

```diff
@@ -131,6 +131,23 @@ export class BasePage {
     );
   }
 
+  /* FUNCTIONALITY: Determines if an Assert has fired. True if yes; false otherwise
+   *
+   * WHY/HOW: Unfortunately, some Asserts don't take focus away from the test app. Because of this, the test execution is able
+   * to continue without realizing an assert as popped up (unless the assert is a crashing one).
+   *
+   * In order to detect an assert, this function gets the number of window handles currently open within the test app.
+   * Only one instance SHOULD be open at a time (the app itself). If another instance of the app is open, we know an assert dialogue
+   * has popped up. With this information, we know to fail the test.
```

Comment:
Are we sure assert dialogs are the only case where we have another window handle open within the test app? I think we have some buttons and links that open other dialogs (like an alert). Those would probably count as a window handle, no?

![image](https://user-images.githubusercontent.com/22876140/152430529-bea9fb08-471b-42a1-8b98-eea007166da1.png)

Reply:
That's a great point. In these tests (we don't have any yet), we can alter the logic a bit to validate another window **does** show up. In all the tests where we **don't** expect a pop-up, we can use the current logic. So it can be on a per-test basis. I don't think there are many situations like this, maybe ~5? But it's a great point to bring up and I'll consider it when I'm writing tests like that.

Reply:
Unfortunately, assert dialogue's are Windows dialogue's and not NetUI. If it was NetUI, we could easily query the Assert itself. But being a Windows handle makes these situations more difficult to handle